prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt

```diff
@@ -164,7 +164,11 @@ class FhirPathCustomResolver(private val customFhirFunctions: FhirPathFunctions?
                 // Convert string constants that are whole integers to Integer type to facilitate math operations
                 values.map {
                     if (it is StringType && StringUtils.isNumeric(it.primitiveValue())) {
-                        IntegerType(it.primitiveValue())
+                        try {
+                            IntegerType(it.primitiveValue())
+                        } catch (e: NumberFormatException) {
+                            it // fallback to string; see https://github.com/CDCgov/prime-reportstream/issues/12609
```

Comment:
Add an extra field in the existing unit test: [test fhir path resolver multiple values](https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/src/test/kotlin/fhirengine/translation/hl7/utils/ConstantResolverTests.kt#L138)

Make the extra field something that triggers the catch block to ensure it falls back to a string.
```kotlin
val bigNumberString = "999999999999999999999999"
```

Reply:
I added a test and adjusted the catch clause to handle the correct exception type (`IllegalArgumentException`) as well as rethrow if the `.cause` is not a `NumberFormatException`.